app/models/video.py

```diff
@@ -1,4 +1,22 @@
 from app import db
+from app.models.rental import Rental
 
 class Video(db.Model):
-    id = db.Column(db.Integer, primary_key=True)
\ No newline at end of file
+    # building the database table with columns, data type, column names
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    release_date = db.Column(db.DateTime) #should this be nullable?
+    total_inventory = db.Column(db.Integer)
+    available_inventory = db.Column(db.Integer, nullable = True)
+    # When the video is deleted, all associated rentals are also deleted
+    # cascade allows us to "delete the ophans" associated with the video id 
+    customers = db.relationship("Rental", backref='Video', cascade="all, delete-orphan", lazy="joined")
+
+    def video_checked_out_count(self):
+        rental_query = Rental.query.filter_by(video_id=self.id, checked_in=False)
+        return rental_query.count()
+
+    def available_inventory(self):
+        available_inv = self.total_inventory - self.video_checked_out_count()
+        return available_inv
```

Comment:
You already have columns in your table for total inventory and available inventory. You do not need to create a function you can just add or subtract 1 inside of the route you need it would look like `video.total_inventory += 1`